Core/Core/Courses/GetCourses.swift

```diff
@@ -102,6 +102,29 @@ public class GetAllCourses: CollectionUseCase {
     public init() {}
 }
 
+public class GetUserCourses: CollectionUseCase {
+    public typealias Model = Course
+
+    let userID: String
+
+    public init(userID: String) {
+        self.userID = userID
+    }
+
+    public var cacheKey: String? { "users/\(userID)/courses" }
+    public var request: GetCoursesRequest {
+        GetCoursesRequest(perPage: 100)
+    }
+
+    public var scope: Scope { Scope(
+        predicate: NSPredicate(format: "ANY %K == %@", #keyPath(Course.enrollments.userID), userID),
```

Comment:
Because enrollments is a set, we want to check if *any* of the elements in the set have the `userID`. If you omit the `ANY` it will crash because enrollments is a many relationship.